src/Tests/Microsoft.NET.TestFramework/ToolsetInfo.cs

```diff
@@ -277,8 +282,19 @@ public static ToolsetInfo Create(string repoRoot, string repoArtifactsDir, strin
 
             if (ret.ShouldUseFullFrameworkMSBuild)
             {
-                // Find path to MSBuildSdkResolver for full framework
-                ret.SdkResolverPath = Path.Combine(repoArtifactsDir, "bin", "Microsoft.DotNet.MSBuildSdkResolver", configuration, "net472", "SdkResolvers");
+                if (repoRoot != null)
+                {
+                    // Find path to MSBuildSdkResolver for full framework
+                    ret.SdkResolverPath = Path.Combine(repoArtifactsDir, "bin", "Microsoft.DotNet.MSBuildSdkResolver", configuration, "net472", "SdkResolvers");
+                }
+                else if (!string.IsNullOrWhiteSpace(commandLine.MsbuildAdditionalSdkRsolverFolder))
+                {
+                    ret.SdkResolverPath = Path.Combine(commandLine.MsbuildAdditionalSdkRsolverFolder, configuration, "net472", "SdkResolvers");
+                }
+                else
+                {
+                    throw new ApplicationException("Microsoft.DotNet.MSBuildSdkResolver path is not provided, set msbuildAdditionalSdkRsolverFolder on test commandline or set repoRoot");
```

Comment:
Nit: I typically use InvalidOperationException for this kind of thing.  I think ApplicationException is sort of soft-deprecated.